multidict/_pair_list.c

```diff
@@ -85,23 +86,68 @@ pair_list_get(pair_list_t *list, Py_ssize_t i)
     return item;
 }
 
-static int
-pair_list_resize(pair_list_t *list, Py_ssize_t new_capacity)
+
+static inline int
+pair_list_grow(pair_list_t *list)
 {
+    // Grow by one element if needed
+    Py_ssize_t new_capacity;
     pair_t *new_pairs;
-    // TODO: use more smart algo for capacity grow
-    if (new_capacity < MIN_LIST_CAPACITY) {
-        new_capacity = MIN_LIST_CAPACITY;
+
+    if (list->size < list->capacity) {
+        return 0;
     }
-    if (list->capacity == new_capacity) {
-        // No need to resize
+
+    if (list->pairs == list->buffer) {
```

Comment:
I don't understand why to compare two ptr? This looks like UB. Mb this should be something like `list->size == list->capacity`?